### ISSUE TITLE
Rule jest-dom/prefer-checked false positive when non-inline arguments passed

### ISSUE THREAD

**Issue by RIP21** [testing-library/eslint-plugin-jest-dom#58]
`eslint-plugin-jest-dom` version: 3.0.1
`node` version: 12.16.1
`npm` version: 6.13.4

This changes 
```javascript
  it.each`
    checked
    ${true}  
    ${false} 
  `('should whatever', ({ checked }) => {
    const { getByTestId } = render(<RenderWhatever checked={checked} />)
    expect(getByTestId(WHATEVER_TEST_ID)).toHaveProperty('checked', checked)
  })
```
to this
```javascript
  it.each`
    checked
    ${true}  
    ${false} 
  `('should whatever', ({ checked }) => {
    const { getByTestId } = render(<RenderWhatever/>)
    expect(getByTestId(WHATEVER_TEST_ID)).not.toBeChecked() 
  })
```
In this scenario, it should ignore fix completely and should not raise a warning about it as it's a valid use case.

**Comment by benmonro** [comment]
Good catch.  I'd welcome a PR if you have cycles

**Comment by RIP21** [comment]
@benmonro sure as we discussed early this morning :D I've just applied fix on 400k+ codebase, so it was almost inevitable I will catch something :D Good for community :D

**Comment by benmonro** [comment]
Ah sorry I'm on my phone didn't realize that was you.  Good stuff!

**Comment by RIP21** [comment]
BTW there is the same behavior for prefer-focus. A bug for sure.

**Comment by benmonro** [comment]
@RIP21 can you please open a separate issue for prefer-focus?  That's a different rule and I could use an example of that for the tests...
